discounts/rust/discounts/default/src/run.rs

```diff
@@ -0,0 +1,391 @@
+use serde::{de::DeserializeOwned, Deserialize};
+
+use serde_json::json;
+use shopify_function::prelude::*;
+use shopify_function::Result;
+
+use cart_fetch::output::{
+    FunctionCartFetchResult, HttpRequest as CartHttpRequest,
+    HttpRequestMethod as CartHttpRequestMethod, HttpRequestPolicy as CartHttpRequestPolicy,
+};
+use cart_run::output::{
+    CartLineTarget, CartOperation, FunctionCartRunResult, OrderDiscountCandidate,
+    OrderDiscountCandidateTarget, OrderDiscountCandidateValue, OrderDiscountSelectionStrategy,
+    OrderDiscounts, OrderSubtotalTarget, Percentage as CartPercentage, ProductDiscountCandidate,
+    ProductDiscountCandidateTarget, ProductDiscountCandidateValue,
+    ProductDiscountSelectionStrategy, ProductDiscounts,
+};
+use delivery_fetch::output::FunctionDeliveryFetchResult;
+use delivery_fetch::output::{
+    HttpRequest as DeliveryHttpRequest, HttpRequestMethod as DeliveryHttpRequestMethod,
+    HttpRequestPolicy as DeliveryHttpRequestPolicy,
+};
+use delivery_run::output::{
+    DeliveryDiscountCandidate, DeliveryDiscountCandidateTarget, DeliveryDiscountCandidateValue,
+    DeliveryDiscountSelectionStrategy, DeliveryDiscounts, DeliveryGroupTarget, DeliveryOperation,
+    FunctionDeliveryRunResult, Percentage as DeliveryPercentage,
+};
+
+type CartResponseData = cart_run::input::ResponseData;
+type DeliveryResponseData = delivery_run::input::ResponseData;
+type CartFetchResponseData = cart_fetch::input::ResponseData;
+type DeliveryFetchResponseData = delivery_fetch::input::ResponseData;
+
+#[derive(Deserialize)]
+struct Metafield {
+    cart_percent: Option<Decimal>,
+    product_percent: Option<Decimal>,
+    delivery_percent: Option<Decimal>,
+}
+
+#[shopify_function_target(
+    target = "cart_run",
+    query_path = "src/run.graphql",
+    schema_path = "schema.graphql"
+)]
+fn cart_run(input: CartResponseData) -> Result<FunctionCartRunResult> {
```

Comment:
This `cart_run` will make:
-  a X% order percent discount on subTotal target
- a Y% product discount on the highest cartLine cost (any better idea?)